lib/gears/async_popen.lua

```diff
@@ -0,0 +1,59 @@
+---------------------------------------------------------------------------
+-- @release @AWESOME_VERSION@
+-- @module gears.async_popen
+---------------------------------------------------------------------------
+
+local setmetatable = setmetatable
+local util = require("awful.util")
+
+local async_popen = { mt = {} }
+
+-- Maps temporary filenames to callbacks
+async_popen.callbacks = {}
+
+--- Run a command asynchronously.
+-- @param command The command to run
+-- @param callback The function to be called when the command exits. The output of the command will be passed to it.
+function async_popen.async_popen(command, callback)
```

Comment:
This could return the result from `util.spawn_with_shell`, which would also include the startup-notification id (`sn`) as of https://github.com/awesomeWM/awesome/commit/f12f9b3b97f2ab23406c3193901fbaf06bccf202.